### PR TITLE
Fix/default markers #85

### LINKED ISSUE #85: Only displaying first geometry item in first collection
The endpoint `http://api.tmpforchch.co.nz/v3/tmps?key=9UdrU5ha&format=geojson` returns 20 geometry collections, of which only the first item(polygon) of the first collection is being displayed when i use the following config file. 

```js
var config = {
  map: {
    domElementId: 'map',
    tileLayers: {
      baseTiles: {
        url: '//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
        attribution: '',
        maxZoom: 18,
        zIndex: 10
      }
    },
    bounds: [
      [-43.577988,172.515934],
      [-43.461397,172.749529]
    ]
  },
  layers: [
    {
      name: 'layer1',
      type: 'geojson',
      dataSource: 'geojson-traffic'
    }
  ],
  dataSources: [
    {
      name: 'geojson-traffic',
      request: {
        url: 'http://api.tmpforchch.co.nz/v3/tmps?key=9UdrU5ha&format=geojson'
      },
      type: 'singlePoll'
    }
  ]
};

var map = Mappy.create(config);
```

### PR DESCRIPTION
**Pull request by markstuart**
Fixes #85

@JonathanPrince, sorry this took sooooo long, but can you please check this out and test it if you get a chance?

Values on the object passed into the `setGeojsonLayer` function on the map wrapper are now 'truthy' (in this case, `undefined` if no relevant config item set, a function otherwise). This allows us to test early whether we should be overriding Leaflet functionality, or using defaults.

### CODE REVIEW

File: wrappers/map.js [hunk 1]
```diff
@@ -59,8 +59,8 @@ class Map extends EventEmitter {
         if (!config.popupFilter || config.popupFilter(feature)) {
 
           // don't bind the feature popup if it isn't defined in config
-          if (typeof config.popup(feature.properties) !== 'undefined') {
-            layer.bindPopup(config.popup(feature.properties))
+          if (typeof config.popupStyle(feature.properties) !== 'undefined') {
+            layer.bindPopup(config.popupStyle(feature.properties))
```
Comment (matt-in-a-hat) [inline wrappers/map.js]: If `popupStyle` isn't set and `notifies` is, will this attempt to call `popupStyle` anyway?

```js
if (config.popupStyle || config.notifies) {
  // ...
  if (typeof config.popupStyle(feature.properties) !== 'undefined') {
```

File: lib/geojsonLayerController.js [hunk 2]
```diff
@@ -34,11 +46,11 @@ function buildGeojsonLayerOptions(options) {
...
-    popup: (properties) => options.popupPresenter.present(properties),
-    popupFilter: (feature) => options.popupFilter.filter(feature),
-    style: (properties) => options.layerStylePresenter.present(properties),
-    icon: (properties) => options.iconStylePresenter.present(properties),
-    filter: (feature)   => options.geojsonFeatureFilter.filter(feature),
+    popupStyle: getStyleFunctionFor('popupPresenter', options),
+    popupFilter: getFilterFunctionFor('popupFilter', options),
+    layerStyle: getStyleFunctionFor('layerStylePresenter', options),
+    iconStyle: getStyleFunctionFor('iconStylePresenter', options),
+    geojsonFilter: getFilterFunctionFor('geojsonFeatureFilter', options),
```
Comment (matt-in-a-hat) [inline lib/geojsonLayerController.js]: `popupStyle:  nn(options)('popupPresenter.present').val` ?

**Comment by matt-in-a-hat** [comment]
Generally looks good, but I haven't tested.

File: wrappers/map.js (hunk 1 above)
Comment (markstuart) [inline wrappers/map.js]: Good call, I'll investigate that one

**Pull request merged**